### DISCUSSION TITLE
Streaming HTTP Response

### DISCUSSION THREAD

**Discussion by albertoe** [scottoffen/grapevine#124] [Q&A]
Is this supported in grapevine?

Is there a sample on how to achieve this?

**Comment by albertoe** [comment]
I have tried the following but it seems to choke

```csharp
using (var stream = new MemoryStream(Encoding.UTF8.GetBytes(responseString)))
{
    context.Response.Headers.Add(HttpRequestHeader.ContentType, "application/json");
    context.Response.Headers.Add(HttpRequestHeader.ContentLength, stream.Length.ToString());
    await context.Response.SendResponseAsync(stream);
}
```

**Reply by scottoffen** [reply]
If you have a string named `responseString` that you want to send as the response:

```csharp
context.Response.ContentType = ContentType.Json;
await content.Response.SendResponseAsync(responseString);
```

**Comment by albertoe** [comment]
My use case calls for streaming of data.
________________________________

**Reply by scottoffen** [reply]
So you want to stream something - e.g. audio or video - is that correct? Or are you wanting to stream something else?

**Reply by albertoe** [reply]
No, I want to stream text.

My use case calls to stream some diagnostic data continuously to clients that request listening to the stream.

**Reply by scottoffen** [reply]
To be honest, I don't know enough about streaming to tell you if it is possible or not at this time. Grapevine was written to be a simple REST/HTTP server, and streaming content was not a use case I had in mind. If anyone has any ideas or suggestions on how to do this, I'd like to hear them.
